### DISCUSSION TITLE
Suggestion for code refactoring regarding check_scene in the Ray Tracing Scene

### DISCUSSION THREAD

**Discussion by jmoraispk** [NVlabs/sionna#189] [General]
Dear Sionna maintainers, 

The functions compute_paths() and compute_coverage_map() of the Scene object have a check_scene parameter that defaults to True and essentially checks whether the scene has properly defined radio parameters. This parameter also has a performance warning flag on its description. 

Checking the scene every time one of these functions is executed is suboptimal, because the verification only needs to be done when the materials change. So, perhaps a cleaner and more efficient implementation would be to have a check_scene() function that is called when the scene changes with regards to its materials. This would allow two nice things:
- increase ease of maintenance by allowing the merge of the methods _ready_for_coverage_map() and _ready_for_paths_computation() which are practically the same.
- eliminate the check_scene parameter and make these functions simpler, offloading from the documentation and user concerns with performance.

That's it. Thank you for your work.

**Comment by jhoydis** [comment]
Hi,

Thanks for your feedback which we will take into account for the development of future releases.
